CMakeLists.txt

```diff
@@ -39,7 +39,8 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
                       -Werror=thread-safety)
 
   if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8)
-    add_compile_options(-Werror=defaulted-function-deleted)
+    add_compile_options(-Werror=defaulted-function-deleted
+                        -Werror=ctad-maybe-unsupported)
```

Comment:
Hopefully it won't get annoying for whatever reason but considering that it was already rarely used it should be fine. I would just add a comment here to explain the decision to disable it.

Reply:
For self-maintained types you always have the option to provide deduction guides and still use CTAD.